sympy/utilities/iterables.py

```diff
@@ -107,15 +107,20 @@ def flatten(iterable, levels=None, cls=None):  # noqa: F811
         else:
             raise ValueError(
                 "expected non-negative number of levels, got %s" % levels)
+        from typing import Any, Optional, Type
+
+        cls: Optional[Type[Any]]
+
+        reducible: Callable[[Any], bool]
```

Comment:
This is not how annotations generally work. Imports go at the top of the file. Parameters should be annotated in the parameter list.